From 72c92875b4f8be13ca2f23d11892fe5fbb238c26 Mon Sep 17 00:00:00 2001
Message-ID: <72c92875b4f8be13ca2f23d11892fe5fbb238c26.1764154288.git.gerald.elder-vass@citrix.com>
In-Reply-To: <61995ad32249580db2c1e899b495a41abfe44d3e.1764154288.git.gerald.elder-vass@citrix.com>
References: <61995ad32249580db2c1e899b495a41abfe44d3e.1764154288.git.gerald.elder-vass@citrix.com>
From: Gerald Elder-Vass <gerald.elder-vass@citrix.com>
Date: Wed, 26 Nov 2025 10:49:36 +0000
Subject: [PATCH 04/12] Revert "hw/i386/pc: Replace
 PCMachineClass::acpi_data_size by PC_ACPI_DATA_SIZE"

This reverts commit e6115657a4437ae13985352ab3e77d78f8cf1ce4.
diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 766360fe96fc..41e1ba5e2a79 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -254,16 +254,6 @@ GlobalProperty pc_compat_2_6[] = {
 };
 const size_t pc_compat_2_6_len = G_N_ELEMENTS(pc_compat_2_6);
 
-/*
- * @PC_FW_DATA:
- * Size of the chunk of memory at the top of RAM for the BIOS ACPI tables
- * and other BIOS datastructures.
- *
- * BIOS ACPI tables: 128K. Other BIOS datastructures: less than 4K
- * reported to be used at the moment, 32K should be enough for a while.
- */
-#define PC_FW_DATA (0x20000 + 0x8000)
-
 GSIState *pc_gsi_create(qemu_irq **irqs, bool pci_enabled)
 {
     GSIState *s;
@@ -648,7 +638,8 @@ void xen_load_linux(PCMachineState *pcms)
     fw_cfg_add_i16(fw_cfg, FW_CFG_NB_CPUS, x86ms->boot_cpus);
     rom_set_fw(fw_cfg);
 
-    x86_load_linux(x86ms, fw_cfg, PC_FW_DATA, pcmc->pvh_enabled);
+    x86_load_linux(x86ms, fw_cfg, pcmc->acpi_data_size,
+                   pcmc->pvh_enabled);
     for (i = 0; i < nb_option_roms; i++) {
         assert(!strcmp(option_rom[i].name, "linuxboot.bin") ||
                !strcmp(option_rom[i].name, "linuxboot_dma.bin") ||
@@ -983,7 +974,8 @@ void pc_memory_init(PCMachineState *pcms,
     }
 
     if (linux_boot) {
-        x86_load_linux(x86ms, fw_cfg, PC_FW_DATA, pcmc->pvh_enabled);
+        x86_load_linux(x86ms, fw_cfg, pcmc->acpi_data_size,
+                       pcmc->pvh_enabled);
     }
 
     for (i = 0; i < nb_option_roms; i++) {
@@ -1784,6 +1776,9 @@ static void pc_machine_class_init(ObjectClass *oc, const void *data)
     pcmc->has_reserved_memory = true;
     pcmc->enforce_amd_1tb_hole = true;
     pcmc->isa_bios_alias = true;
+    /* BIOS ACPI tables: 128K. Other BIOS datastructures: less than 4K reported
+     * to be used at the moment, 32K should be enough for a while.  */
+    pcmc->acpi_data_size = 0x20000 + 0x8000;
     pcmc->pvh_enabled = true;
     pcmc->kvmclock_create_always = true;
     x86mc->apic_xrupt_override = true;
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index a84d5b502a76..4afd2b686313 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -79,6 +79,9 @@ typedef struct PCMachineState {
  *
  * Compat fields:
  *
+ * @acpi_data_size: Size of the chunk of memory at the top of RAM
+ *                  for the BIOS ACPI tables and other BIOS
+ *                  datastructures.
  * @gigabyte_align: Make sure that guest addresses aligned at
  *                  1Gbyte boundaries get mapped to host
  *                  addresses aligned at 1Gbyte boundaries. This
@@ -102,6 +105,7 @@ struct PCMachineClass {
 
     /* ACPI compat: */
     bool has_acpi_build;
+    unsigned acpi_data_size;
     int pci_root_uid;
 
     /* SMBIOS compat: */
